src/transforms/key_value_parser.rs

```diff
@@ -0,0 +1,400 @@
+#[serde(default, deny_unknown_fields)]
+pub struct KeyValueConfig {
+    #[derivative(Default(value = "true"))]
+    pub drop_field: bool,
+    pub field: Option<Atom>,
+    pub field_split: Option<String>,
+    #[derivative(Default(value = "true"))]
+    pub overwrite_target: bool,
+    pub separator: Option<String>,
+    pub target_field: Option<Atom>,
+    pub trim_key: Option<String>,
+    pub trim_value: Option<String>,
+    pub types: HashMap<Atom, String>,
+}
+
+inventory::submit! {
+    TransformDescription::new::<KeyValueConfig>("key_value_parser")
+}
+
+#[typetag::serde(name = "key_value_parser")]
+impl TransformConfig for KeyValueConfig {
+    fn build(&self, _cx: TransformContext) -> crate::Result<Box<dyn Transform>> {
+        let conversions = parse_conversion_map(&self.types)?;
+        let field = self.field.as_ref().unwrap_or(&log_schema().message_key());
+        let field_split = self.field_split.clone().unwrap_or_else(|| "=".to_string());
+        let separator = self.separator.clone().unwrap_or_else(|| " ".to_string());
+        let trim_key = self.trim_key.as_ref().map(|key| key.chars().collect());
+        let trim_value = self.trim_value.as_ref().map(|key| key.chars().collect());
+
+        // Ensure the field being dropped is not the target field.
+        let drop_field = self.drop_field
+            && self
+                .target_field
+                .as_ref()
+                .map(|target_field| field != target_field)
+                .unwrap_or(true);
+
+        Ok(Box::new(KeyValue {
+            conversions,
+            drop_field,
+            field: field.clone(),
+            field_split,
+            overwrite_target: self.overwrite_target,
+            separator,
+            target_field: self.target_field.clone(),
+            trim_key,
+            trim_value,
+        }))
+    }
+
+    fn input_type(&self) -> DataType {
+        DataType::Log
+    }
+
+    fn output_type(&self) -> DataType {
+        DataType::Log
+    }
+
+    fn transform_type(&self) -> &'static str {
+        "key_value_parser"
+    }
+}
+
+pub struct KeyValue {
+    conversions: HashMap<Atom, Conversion>,
+    drop_field: bool,
+    field: Atom,
+    field_split: String,
+    overwrite_target: bool,
+    separator: String,
+    target_field: Option<Atom>,
+    trim_key: Option<Vec<char>>,
+    trim_value: Option<Vec<char>>,
+}
+
+impl KeyValue {
+    fn parse_pair(&self, pair: &str) -> Option<(Atom, String)> {
+        let pair = pair.trim();
+        let field_split = &self.field_split;
+
+        let fields = if field_split.is_empty() {
+            let mut kv_pair = pair.split_whitespace();
+            let key = kv_pair.next()?;
+            let val = kv_pair.next()?;
+            if kv_pair.next().is_some() {
+                error!(
+                    message = "KeyValue parser saw more than one separator",
+                    rate_limit_secs = 30
+                );
```

Comment:
I almost removed this functionality. It's a weird-ish edge case to want to split a field on white space. I considered converting empty string inputs to the default `=`. I figured it was probably best to not restrict a user if they _really_ wanted  to do this for some reason.

Reply:
If splitting a key+value on white space is that low value, it would be best to remove this as the default. I don't see it being useful myself. Defaulting to `"="` as you suggest seems to make more sense.

Reply:
I've updated it to remove whitespace as an option for field_split and default to `=`.